doc/classes/StringName.xml

```diff
@@ -5,7 +5,7 @@
 	</brief_description>
 	<description>
 		[StringName]s are immutable strings designed for general-purpose representation of unique names (also called "string interning"). Two [StringName]s with the same value are the same object. Comparing them is extremely fast compared to regular [String]s.
-		You will usually just pass a [String] to methods expecting a [StringName] and it will be automatically converted, but you may occasionally want to construct a [StringName] ahead of time with the [StringName] constructor or, in GDScript, the literal syntax [code]&amp;"example"[/code].
+		You will usually pass a [String] to methods expecting a [StringName] and it will be automatically converted (often at compile time), but in rare cases you can construct a [StringName] ahead of time with the [StringName] constructor or, in GDScript, the literal syntax [code]&amp;"example"[/code]. Manual construction allows you to control when [StringName] conversions occur (moving them out of loops) or eliminate them (using the literal).
```

Comment:
(Feel free to tweak slightly)
```suggestion
		You will usually pass a [String] to methods expecting a [StringName] and it will be automatically converted (often at compile time), but in rare cases you can construct a [StringName] ahead of time with the [StringName] constructor or, in GDScript, the literal syntax [code]&amp;"example"[/code]. Manually defining a [StringName] allows you to control when the conversion from [String] occurs or prevent it entirely, which can be useful for loops or other performance-expensive operations.
```

Reply:
I'd use `declaring` over `defining` here, or just `constructing`

Reply:
Edited the suggestion accordingly so that they can just copy it when they see it.

Reply:
Changed to 'constructing'.

I'm reluctant to vaguely connect "Manually constructing" with "performance-expensive operations" because people tend to simplify to "construction is faster, always do that" instead of "strings are usually fine except in these rare cases". And as a user, it's helpful to know exactly when it's more performant. 

Also, I don't think it's entirely clear "prevent" means "use the literal" without mentioning the literal.

Resolving. But if this would prevent merging, please re-open.